CONTRIBUTING.md

```diff
@@ -28,65 +28,116 @@ Here are questions you can answer before you file a bug to make sure you're not
+
+Tags in titles allow for speedy categorization
+Title tags  are generally one word or acronym enclosed in square brackets.
+Limiting to one or two tags is ideal to keep titles short.
+Some examples of common tags are:
+
+- `[NFC]` - No Functional Change
+- `[RFC]` - Request For Comments (often used for drafts to get feedback)
+- `[Doc]` - Documentation change
+- `[SPIRV]` - Changes related to SPIR-V
+- `[HLSL2021]` - Changes related to HLSL 2021 features
+- Other tags in use: `[Linux]`, `[mac]`, `[Win]`, `[PIX]`, etc...
+
+Tags aren't formalized or any specific limited set.
+If you're unsure of a reasonable tag to use, just don't use any.
+If you want to invent a new tag, go for it!
+These are to help categorize changes at a glance.
+
+#### Descriptions
+
+The PR description should include a more detailed description of the change,
+ an explanation for the motivation of the change, and links to any relevant Issues.
+This does not need to be a dissertation, but should leave breadcrumbs for the next person debugging your code (who might be you).
+
+Using the words `Fixes`, `Fixed`, `Closes`, `Closed`, or `Close` followed by
+ `#<issuenumber>`, will auto close an issue after the PR is merged.
+
+#### Release Notes
+
+Significant changes may require release notes that highlight important
+ compiler behavior changes for each named release.
+These include changes that are:
+
+- Visible to the users
+- Significant changes to compiler behavior
+  - New features: Language, Hardware support, compiler options
+  - Important bug fixes
+  - Changes in default behavior
+
+When such a change is made, the release note should be included as part of that change.
+This is done in the docs/ReleaseNotes.md file.
+
+If the change is meant for a named release, it should be added to that named release's section of the release notes file.
+As the change is merged to the appropriate release branches, the release notes will come along with it.
+
+If a change is meant for the next upcoming release, it should be added to the "Upcoming Release" section.
+When the next upcoming release is named, the title will be updated and the release note will be included in the appropriate release.
+
+When writing release note list entries:
+
+- Keep the description to a single sentence.
+- Links to specific PRs shouldn't be included.
+- Markdown links to bugs are encouraged if the issue is too complicated to completely explain in a single sentence.
+- Remember to update release notes as the nature of the change alters or is removed.
+
+### Testing Pull Requests
+
+All changes that make functional or behavioral changes to the compiler whether by fixing bugs or adding features
+ must include additional testing in the implementing pull request.
+Changes that do not change behavior may still be required to add testing if the change impacts areas with limited test coverage
+ to verify that the change doesn't alter previously untested, but important behavior.
```

Comment:
nit: couple more odd line wraps here.

Reply:
These are all intended. I find shorter lines easier to review and the space on the next line is meant to indicate a continuation of the previous line. When it's processed, both the newline and the space get converted into a single space. 

It's not a hill I'm determined to die on, but I think it's helpful for reviewing the source.

Reply:
I think if you want line wrapping it should be consistent. This is crazy inconsistent across the whole file.